testsuite/MDAnalysisTests/core/test_atomselections.py

```diff
@@ -372,6 +372,14 @@ def test_global(self, universe):
     def test_accept_atomgroup(self, universe, selstr):
         ag = universe.select_atoms(selstr)
         assert ag is universe.select_atoms(ag)
+    
+    @pytest.mark.parametrize('selstr',
+        ["around 4.0 bynum 1943",
+        "around 10.0 bynum 1943"
+        ])
+    def test_accept_updatingatomgroup(self, universe, selstr):
+        ag = universe.select_atoms(selstr, updating=True)
+        assert ag is universe.select_atoms(ag)
```

Comment:
So the ‘is’ statement here checks that they are literally the same object, but I think there’s (possible) implementations where it returns a copy. So the test might be better with ==

Reply:
I'm not sure I understand what you mean -- can checking `isinstance` modify the UpdatingAtomGroup?

https://github.com/MDAnalysis/mdanalysis/blob/cc71ab887bf5ee4609adef48a5ba8936ee65e963/package/MDAnalysis/core/groups.py#L2841-L2842

Reply:
The test is using “assert A is B” not “assert A == B”, these are slightly different

Reply:
Perhaps have two tests: one that checks that you get the same object back (`is`) the other that it "is equal" (`==`). In this way we get noticed if our code does a copy somewhere where we don't expect it.